src/LineageTree/lineageTree.py

```diff
@@ -78,113 +261,177 @@ def add_node(
+    #     for t in times:
+    #         self.time_nodes[t] = set(self.time_nodes[t]).difference(track)
+    #     for i, c in enumerate(track):
+    #         self.pos.pop(c)
+    #         if i != 0:
+    #             self.predecessor.pop(c)
+    #         if i < len(track) - 1:
+    #             self.successor.pop(c)
+    #         self.time.pop(c)
 
-        Args:
-            c (int): id of the node to remove
-        """
-        self.nodes.remove(c)
-        self.time_nodes[self.time[c]].remove(c)
-        # self.time_nodes.pop(c, 0)
-        pos = self.pos.pop(c, 0)
-        e_to_remove = [e for e in self.edges if c in e]
-        for e in e_to_remove:
-            self.edges.remove(e)
-        if c in self.roots:
-            self.roots.remove(c)
-        succ = self.successor.pop(c, [])
-        s_to_remove = [s for s, ci in self.successor.items() if c in ci]
-        for s in s_to_remove:
-            self.successor[s].remove(c)
-
-        pred = self.predecessor.pop(c, [])
-        p_to_remove = [s for s, ci in self.predecessor.items() if ci == c]
-        for s in p_to_remove:
-            self.predecessor[s].remove(c)
-
-        self.time.pop(c, 0)
-        self.spatial_density.pop(c, 0)
-
-        self.next_id.append(c)
-        return e_to_remove, succ, s_to_remove, pred, p_to_remove, pos
-
-    def fuse_nodes(self, c1: int, c2: int):
-        """Fuses together two nodes that belong to the same time point
-        and update the lineageTree accordingly.
+    def remove_nodes(self, group):
+        """Removes a group of nodes from the LineageTree
 
         Args:
-            c1 (int): id of the first node to fuse
-            c2 (int): id of the second node to fuse
+            group (set|int): One or more nodes that are to be removed.
         """
-        (
-            e_to_remove,
-            succ,
-            s_to_remove,
-            pred,
-            p_to_remove,
-            c2_pos,
-        ) = self.remove_node(c2)
-        for e in e_to_remove:
-            new_e = [c1] + [other_c for other_c in e if e != c2]
-            self.edges.add(new_e)
+        if isinstance(group, int):
+            group = {group}
+        self.nodes.difference_update(group)
+        times = {self.time.pop(n) for n in group}
+        for t in times:
+            self.time_nodes[t] = set(self.time_nodes[t]).difference(group)
+        for node in group:
+            self.pos.pop(node)
+            if self.predecessor.get(node):
+                pred = self.predecessor[node][0]
+                siblings = self.successor.pop(pred, [])
+                siblings = [] if siblings is None else siblings
+                if len(siblings) == 2:
+                    siblings.remove(node)
+                    self.successor[pred] = siblings
+                self.predecessor.pop(node)
+            if self.successor.get(node):
+                for succ in self.successor[node]:
+                    self.predecessor.pop(succ)
```

Comment:
Here you should be able to do that:
```python
for succ in self.successor.get(node, []):
    self.predecessor.pop(succ)
```
also `if self.successor.get(node)` is equivalent to `if node in self.successor`.
I find the second option better